### PR TITLE
Add PyPI publishing CD workflow

### PR DESCRIPTION
**Pull request by vbaltrusaitis-reef**
- Add GitHub continuous delivery workflow which is triggered on `library-vX.X.X` tag. The workflow
  - Creates a GitHub release.
  - Pushes the package to PyPI using [trusted publisher flow](https://docs.pypi.org/trusted-publishers/) (no token in secrets necessary).
- Add `noxfile.py` that can format, lint, and make a release commit (writes [towncrier](https://towncrier.readthedocs.io/en/stable/tutorial.html) log items to `/compute_horde/CHANGELOG.md`).
- Change Docker image build scripts to use additional context to copy `compute_horde` package instead or relying on copying (as far as I can tell, docker-compose would not work before running `./build-image.sh` before this because it needed a `compute_horde` subdirectory within `/validator`, `/miner`, `/executor`).

Note that while this implements publishing, miner, validator and executor projects still use path `../compute_horde` as a dependency. Changing this would have some effects on DX that I'd like discuss more first.

### CODE REVIEW

**Comment by vbaltrusaitis-reef** [comment]
To be fair, this needs more documentation, a proper `/compute_horde/README.md` and a `/CONTRIBUTING.md`.

File: executor/app/build-image.sh [hunk 1]
```diff
@@ -1,6 +1,5 @@
 #!/bin/bash -eux
 
 IMAGE_NAME="backenddevelopersltd/compute-horde-executor:v0-latest"
-rsync -avzP ../../compute_horde packages/
 # Dockerfile assumes that build is run in the parent dir
-cd .. && docker build -t $IMAGE_NAME -f app/envs/prod/Dockerfile .
+cd .. && docker build --build-context compute-horde=../compute_horde -t $IMAGE_NAME -f app/envs/prod/Dockerfile .
```
Comment (agoncharov-reef) [inline executor/app/build-image.sh:5]: Game changer!

File: miner/app/build-image.sh [hunk 2]
```diff
@@ -1,6 +1,5 @@
 #!/bin/bash -eux
 
 IMAGE_NAME="backenddevelopersltd/compute-horde-miner:v0-latest"
-rsync -avzP ../../compute_horde packages/
 # Dockerfile assumes that build is run in the parent dir
-cd .. && docker build -t $IMAGE_NAME -f app/envs/prod/Dockerfile .
+cd .. && docker build --build-context compute-horde=../compute_horde -t $IMAGE_NAME -f app/envs/prod/Dockerfile .
```
Comment (agoncharov-reef) [inline miner/app/build-image.sh:5]: Game changer!

File: miner/envs/prod/docker-compose.yml [hunk 3]
```diff
@@ -33,6 +33,8 @@ services:
     build:
       context: .
       dockerfile: app/Dockerfile
+      additional_contexts:
+        compute-horde: ../compute_horde
```
Comment (agoncharov-reef) [inline miner/envs/prod/docker-compose.yml:37]: Game changer!

File: validator/app/build-image.sh [hunk 4]
```diff
@@ -1,6 +1,5 @@
 #!/bin/bash -eux
 
 IMAGE_NAME="backenddevelopersltd/compute-horde-validator:v0-latest"
-rsync -avzP ../../compute_horde packages/
 # Dockerfile assumes that build is run in the parent dir
-cd .. && docker build -t $IMAGE_NAME -f app/envs/prod/Dockerfile .
+cd .. && docker build --build-context compute-horde=../compute_horde  -t $IMAGE_NAME -f app/envs/prod/Dockerfile .
```
Comment (agoncharov-reef) [inline validator/app/build-image.sh:5]: Game changer!

File: validator/envs/prod/docker-compose.yml [hunk 5]
```diff
@@ -33,6 +33,8 @@ services:
     build:
       context: .
       dockerfile: app/Dockerfile
+      additional_contexts:
+        compute-horde: ../compute_horde
```
Comment (agoncharov-reef) [inline validator/envs/prod/docker-compose.yml:37]: Game changer!

**Review by agoncharov-reef — APPROVED** [review]
Cool stuff!

**Pull request merged**